### ISSUE TITLE
example_recursive.rb breaks .xlsx file

### ISSUE THREAD

**Issue by dukedave** [rubyzip/rubyzip#343]
When opening a `.xlsx` file zipped via `example_recursive.rb` in Excel 15.39 on Mac OS, I am presented with:

<img width="430" alt="broken-xlsx" src="https://user-images.githubusercontent.com/48232/32582873-2e7c52ce-c4ae-11e7-8760-7d8af58ea6da.png">

Clicking through to the error log I see:

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<recoveryLog xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main">

<logFileName>Repair Result to SOME_FILE.xml</logFileName><summary>Errors were detected in file 'SOME_FILE.xlsx'</summary>

<additionalInfo><info>Excel completed file level validation and repair. Some parts of this workbook may have been repaired or discarded.</info></additionalInfo>

</recoveryLog>
```

Not fixed by #260, not sure if related to https://github.com/rubyzip/rubyzip#modify-docx-file-with-rubyzip ?

**Comment by dukedave** [comment]
My files were nested at most one folder deep, so I just rolled by own version, which worked first time. Only obvious difference is that I used `zip_file.add file_name, file_path`, whereas `example_recursive.rb` uses `get_output_stream` and `write`.

**Comment by lanrion** [comment]
+1

**Comment by qwlong** [comment]
+1

**Comment by jdleesmiller** [comment]
I think this was fixed in #297, which used the approach mentioned in https://github.com/rubyzip/rubyzip/issues/343#issuecomment-343013460 .
